Visualisation/js/attributetable.js

```diff
@@ -0,0 +1,248 @@
+	.attr("x", widthSong/2)
+	.attr("y", heightElement/2)
+    .attr("dominant-baseline", "central");
+
+for (let i = 0; i < attributes.length; i++) {
+    svgTable.append("rect")
+        .attr("id", attributes[i].id + "Headline")    
+        .attr("x", widthSong + ((width - widthSong)/attributes.length) * i )
+        .attr("y", 0)
+        .attr("width", ((width - widthSong)/attributes.length))
+        .attr("height", heightElement)
+        .attr("fill", "none")
+        .attr("stroke-width", 2)
+        .attr("stroke", "black");
+    svgTable.append("text")
+        .style("text-anchor", "middle")
+        .text(attributes[i].attribute)
+        .attr("class", "tableHeadline")
+        .attr("x", widthSong + ((width - widthSong)/attributes.length) * i + ((width - widthSong)/attributes.length)/2)
+        .attr("y", heightElement/2)
+        .attr("dominant-baseline", "central");
+    };
+
+// making the table grid and append the values to the table
+ 
+for (let i = 0 ; i < tableData.length; i++){
+    // adding box for higlighting a row when on hover
+    svgTable.append("rect")
+        .attr("id", "Row" + i + "highlight")
+        .attr("x", 0)
+        .attr("y", heightElement + (heightElement * i))
+        .attr("width", width)
+        .attr("height", heightElement)
+        .attr("fill", "none");
+    
+    svgTable.append("text")
+        .attr("id", "song" + i + "inTable")
+        .attr("x", 5)
+	    .attr("y", heightElement + heightElement *i + heightElement/4)
+        .attr("dominant-baseline", "central")
+        .html(`<tspan class="card-title" dy="0">${cropText(svgTable, tableData[i][1], widthSong)}</tspan><br><tspan class="card-text" x="5" dy="1.2em">${cropText(svgTable, tableData[i][2], widthSong)}</tspan>`);
+    
+    svgTable.append("rect")
+        .attr("class", "songInTable Row" + i)
+        .attr("x", 0)
+        .attr("y", heightElement + heightElement *i)
+        .attr("width", widthSong)
+        .attr("height", heightElement)
+        .attr("fill-opacity", 0)
+        .attr("stroke-width", 2)
+        .attr("stroke", "black")
+        .on("mouseover", function() {
+            // On hover, change the outline to a different color
+            d3.select("#Row" + i + "highlight" ).attr("stroke", tableData[i][11]).attr("stroke-width", 4).style("visibility", "visible");
+        })
+        .on("mouseout", function() {
+            // On mouse out, revert the outline color
+            d3.select("#Row" + i + "highlight").style("visibility", "hidden"); // Revert back to black
+        });
+    
+    
+
+        for (let j = 0; j < attributes.length; j++){
+    svgTable.append("rect")
+        .attr("class", attributes[j].id + " Row" +i)    
+        .attr("x", widthSong + ((width - widthSong)/attributes.length) * j )
+        .attr("y", heightElement + heightElement *i)
+        .attr("width", ((width - widthSong)/attributes.length))
+        .attr("height", heightElement)
+        .attr("fill", "black")
```

Comment:
I think it would be very nice to use the song colors here instead of black. this way, its even easier to spot what attributes belong to what song. If you do this, perhaps the on-hover border should be black. dont be afraid to @ me if you disagree :p

Reply:
@KoSchoemaker I disagree with different colours in one column. With that we get a different channel in the visualisation that also makes it difficult to compare the values in one column to each other